File: app/src/main/java/io/github/zwieback/familyfinance/business/iconics/fragment/item/SpaceItemDecoration.java
package io.github.zwieback.familyfinance.business.iconics.fragment.item;

import android.content.res.Resources;
import android.graphics.Rect;
import android.view.View;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.RecyclerView.State;

/**
 * @author pa.gulko zTrap (28.10.2017)
 */
public class SpaceItemDecoration extends RecyclerView.ItemDecoration {

    @Override
    public void getItemOffsets(
            @NonNull Rect outRect,
            @NonNull View view,
            @NonNull RecyclerView parent,
            @NonNull State state
    ) {
        int space = (int) Math.ceil(8 * Resources.getSystem().getDisplayMetrics().density); // 8 dp
        int position = parent.getChildAdapterPosition(view);
        if (position < 2) {
            outRect.top = space;
        }
        if ((position + 1) % 2 == 0) {
            outRect.right = space;
        }
        outRect.left = space;
        outRect.bottom = space;
    }
}

File: app/src/main/java/io/github/zwieback/familyfinance/business/iconics/fragment/item/SpaceItemDecoration.kt
package io.github.zwieback.familyfinance.business.iconics.fragment.item

import android.content.res.Resources
import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.State
import kotlin.math.ceil

/**
 * @author pa.gulko zTrap (28.10.2017)
 */
class SpaceItemDecoration : RecyclerView.ItemDecoration() {

    override fun getItemOffsets(
        outRect: Rect,
        view: View,
        parent: RecyclerView,
        state: State
    ) {
        val space =
            ceil((8 * Resources.getSystem().displayMetrics.density).toDouble()).toInt() // 8 dp
        val position = parent.getChildAdapterPosition(view)
        if (position < 2) {
            outRect.top = space
        }
        if ((position + 1) % 2 == 0) {
            outRect.right = space
        }
        outRect.left = space
        outRect.bottom = space
    }
}
